lms/djangoapps/grades/new/subsection_grade.py

```diff
@@ -223,20 +240,22 @@ def create(self, subsection, block_structure=None, read_only=False):
         if not subsection_grade:
             subsection_grade = SubsectionGrade(subsection, self.course)
             subsection_grade.compute(self.student, block_structure, self._scores_client, self._submissions_scores)
-            if PersistentGradesEnabledFlag.feature_enabled(self.course.id):
-                if read_only:
-                    self._unsaved_subsection_grades.append(subsection_grade)
-                else:
-                    grade_model = subsection_grade.create_model(self.student)
-                    self._update_saved_subsection_grade(subsection.location, grade_model)
+            with persistence_safe_fallback():
```

Comment:
Can we move this to within the else clause below?  The other path doesn't update the database.